File: fastcampus/05.그래프 탐색/1260.py
import sys
input = sys.stdin.readline

from collections import deque

n, m, v = map(int, input().split())
graph = [[] for _ in range(n+1)]
visited = [0 for _ in range(n+1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

for el in graph:
    el.sort()

def bfs(start):
    q = deque()
    visit = [0 for _ in range(n+1)]
    q.append(start)
    visit[start] = 1

    while q:
        now = q.popleft()
        print(now, end = ' ')
        for ne in graph[now]:
            if visit[ne] == 0:
                q.append(ne)
                visit[ne] = 1

def dfs(start):
    print(start, end = " ")
    visited[start] = 1
    for ne in graph[start]:
        if visited[ne] == 0:
            dfs(ne)

dfs(v)
print()
bfs(v)

File: solved.ac/CLASS3_JS/1260.js
const fs = require("fs");
const inputPath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const stdin = fs
  .readFileSync(inputPath)
  .toString()
  .split("\n")
  .map((line) => line.trim());

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const dfs = (graph, node, visited, ans) => {
  ans.push(node);
  for (let next_ of graph[node]) {
    if (visited[next_] === 0) {
      visited[next_] = 1;
      dfs(graph, next_, visited, ans);
    }
  }
};

const bfs = (graph, start, visited) => {
  const q = [[start]];
  const ans = [];

  while (q.length) {
    const node = q.shift();
    ans.push(node);
    for (let next_ of graph[node]) {
      if (visited[next_] === 0) {
        visited[next_] = 1;
        q.push([next_]);
      }
    }
  }
  return ans.join(" ");
};

const solution = () => {
  const [n, m, v] = input().split(" ").map(Number);
  const graph = Array.from({ length: n + 1 }, () => []);
  const ans = [];

  for (let i = 0; i < m; i++) {
    const [u, v] = input().split(" ").map(Number);
    graph[u].push(v);
    graph[v].push(u);
  }

  for (let i = 0; i < graph.length; i++) {
    graph[i].sort((a, b) => a - b);
  }
  let visited = Array(n + 1).fill(0);
  visited[v] = 1;

  dfs(graph, v, visited, ans);
  console.log(ans.join(" "));

  visited = Array(n + 1).fill(0);
  visited[v] = 1;
  console.log(bfs(graph, v, visited));
};

solution();
